### PR TITLE
hssi 10G: port_ -> src_port_ and dst_port_

### PR DESCRIPTION
**Pull request by tswhison**
Replaces the notion of a single port for both Tx/Rx with one port for the Tx side and a separate port for the Rx side.

Adds mbox_write(), mbox_read() variants that accept a port selection as a parameter.

### CODE REVIEW

**Review by anandaravuri — DISMISSED** [review]
Can you add Tx and Rx Port changes to 100G also ?

**Comment by tswhison** [comment]
Yes, but I'd like to save that for a separate pull request.

File: samples/hssi/hssi_10g_cmd.h [hunk 1]
```diff
@@ -159,8 +170,13 @@ class hssi_10g_cmd : public hssi_cmd
     if (eth_ifc_ == "none")
         eth_ifc = hafu->ethernet_interface();
 
+    if (port_ != INVALID_PORT)
```
Comment (anandaravuri) [inline samples/hssi/hssi_10g_cmd.h:173]: corner case:  if user pass all port_, dst_port_,  src_port_ values  HSSI tool overrides  dst_port_,  src_port_ values.

Comment (tswhison) [reply samples/hssi/hssi_10g_cmd.h:173]: Yes, would you like me to also print a message in this case?

Comment (anandaravuri) [reply samples/hssi/hssi_10g_cmd.h:173]: Print message and continue

**Review by anandaravuri — DISMISSED** [review]

File: samples/hssi/hssi_10g_cmd.h [hunk 2]
```diff
@@ -352,20 +356,20 @@ class hssi_10g_cmd : public hssi_cmd
     os << "0x3c0d " << std::setw(22) << "pkt_length" << ": " <<
       int_to_hex(hafu->mbox_read(CSR_PACKET_LENGTH)) << std::endl;
     os << "0x3cf4 " << std::setw(22) << "tx_end_tstamp" << ": " <<
-      int_to_hex(hafu->mbox_read(CSR_TX_END_TSTAMP)) << std::endl;
+      int_to_hex(hafu->mbox_read(src_port_, CSR_TX_END_TSTAMP)) << std::endl;
   
     os << "0x3d00 " << std::setw(22) << "num_pkt" << ": " <<
-      int_to_hex(hafu->mbox_read(CSR_NUM_PKT)) << std::endl;
```
Comment (aditya-patro) [inline samples/hssi/hssi_10g_cmd.h:358]: 0x3d00 to 0x3dff all csr access are for dst_port

Comment (tswhison) [reply samples/hssi/hssi_10g_cmd.h:358]: Got it, thanks! 👍

**Review by anandaravuri — APPROVED** [review]

**Review by anandaravuri — DISMISSED** [review]

**Review by anandaravuri — APPROVED** [review]

**Pull request merged**